src/types.ts

```diff
@@ -78,10 +78,15 @@ export interface S3Config {
   readonly authenticationRequired?: boolean;
 }
 
+export interface UnauthorizedResponseHeaders {
+  readonly accessControlAllowOrigin?: string;
+}
+
 export interface BasicAuthenticationConfig {
   readonly username: string;
   readonly password: string;
   readonly cacheTtlInSeconds?: number;
+  readonly unauthorizedResponseHeaders?: UnauthorizedResponseHeaders;
```

Comment:
maybe rename this or add a comment, to make clear that this is mainly for the initial request

Reply:
this is not only for the initial request, it is also for 401s where the credentials are wrong

Reply:
mainly vs only ;)

Reply:
So what is you suggestion here? I'm not sure what needs explaining.

Reply:
I added a comment within the `UnauthorizedResponseHeaders` interface to the `accessControlAllowOrigin` property.